Source/AppleSocial.swift

```diff
@@ -0,0 +1,119 @@
+//
+//  AppleSocial.swift
+//  edX
+//
+//  Created by Saeed Bashir on 8/24/20.
+//  Copyright © 2020 edX. All rights reserved.
+//
+
+import Foundation
+import UIKit
+import MSAL
+import AuthenticationServices
+
+typealias AppleLoginCompletionHandler =  (OEXRegisteringUserDetails?, _ accessToken: String?, Error?) -> Void
+
+
+class AppleSocial: NSObject {
+
+    static let shared = AppleSocial()
+    var completionHandler: AppleLoginCompletionHandler!
```

Comment:
`completionHandler` can be made optional and private
```suggestion
    private var completionHandler: AppleLoginCompletionHandler?
```